### ISSUE TITLE
[BigCommerce] Logged in users error when checking out

### ISSUE THREAD

**Issue by markkdev** [vercel/commerce#676]
Hi,

Love the work you're all doing here and appreciate your effort.

I'm almost ready to launch a store, i'm having an issue with checkout. 

**TLDR**: how do we get the client secret that's created by the initial Vercel deployment?

I'm running into an issue where I've created my own API keys in my BC store. I've used these keys to create a graphql storefront token.

**Locally**

Checkout works great. I'm able to get redirected to my store while still being logged in.

**[In CI]**

I'm running into two issues.

The first, if I create a storefront_api_token with the api credentials **I've** created, the build fails with an authentication error. I assume this is because of CORS.

What allowed CORS origin should be set for vercel? wild cards aren't allowed by bigcommerce so `https://*.vercel.app` doesn't work. I also tried `https://vercel.app` no luck.

The second, is it possible to just get the CLIENT_SECRET created by the integration? Everything would work with this client secret but it doesnt come down when you `vercel env pull`

CLIENT_SECRET seems to be never created because when you sign the jwt for login -> checkout redirect it throws an error due to the fact that it isnt created. 

If you use a STOREFRONT_API_TOKEN that was created by a different set of API credentials, login to bigcommerce fails and the logged in user is never redirected to checkout until the login again.

**Comment by markkdev** [comment]
Shortly after this I think I understand whats happening.

I was under the assumption that when creating a storefront_api_token the body of the `POST` request to the bigcommerce API **must** look like this. 

`{
  "channel_id": 1,
  "expires_at": 1234567890,
  "allowed_cors_origins": [...]
}`

however it seems that if you completely remove the `allowed_cors_origins` field it will still allow you to create a token. 

My request when creating a token is now just 

`{
  "channel_id": 1,
  "expires_at": 1234567890,
}`

I've also had to update some of the environment variables such as store_url etc according to [this guide](https://developer.bigcommerce.com/api-docs/storefronts/nextjs-commerce)

Closing this issue as it seems to be solved. Hope its helpful to someone else that runs into this problem. 

Cheers
